src/BDInfo/TSPlaylistFile.cs

```diff
@@ -17,1346 +17,1150 @@
             }
+            return 0;
         }
+    }
 
-        public ulong TotalAngleBitRate
+    public ulong TotalAngleBitRate
+    {
+        get
         {
-            get
+            if (TotalAngleLength > 0)
             {
-                if (TotalAngleLength > 0)
-                {
-                    return (ulong)Math.Round(TotalAngleSize * 8.0 / TotalAngleLength);
-                }
-                return 0;
+                return (ulong)Math.Round(TotalAngleSize * 8.0 / TotalAngleLength);
             }
+            return 0;
         }
+    }
 
-        public string GetFilePath()
-        {
-            if (string.IsNullOrEmpty(FileInfo?.FullName))
-            {
-                return string.Empty;
-            }
+    public string GetFilePath()
+    {
+        return !string.IsNullOrEmpty(_fileInfo.FullName) ? _fileInfo.FullName : string.Empty;
+    }
 
-            return FileInfo.FullName;
-        }
+    public void Scan(Dictionary<string, TSStreamFile> streamFiles, Dictionary<string, TSStreamClipFile> streamClipFiles)
+    {
+        Stream fileStream = null;
+        BinaryReader fileReader = null;
 
-        public void Scan(
-            Dictionary<string, TSStreamFile> streamFiles,
-            Dictionary<string, TSStreamClipFile> streamClipFiles)
+        try
         {
-            Stream fileStream = null;
-            BinaryReader fileReader = null;
-            ulong streamLength = 0;
+            Streams.Clear();
+            StreamClips.Clear();
 
-            try
-            {
-                Streams.Clear();
-                StreamClips.Clear();
-
-                fileStream = FileInfo.OpenRead();
-                fileReader = new BinaryReader(fileStream);
-                streamLength = (ulong)fileStream.Length;
+            fileStream = _fileInfo.OpenRead();
+            fileReader = new BinaryReader(fileStream!);
+            var streamLength = (ulong)fileStream.Length;
 
-                byte[] data = new byte[streamLength];
-                int dataLength = fileReader.Read(data, 0, data.Length);
+            var data = new byte[streamLength];
+            var dataLength = fileReader.Read(data, 0, data.Length);
 
-                int pos = 0;
+            var pos = 0;
 
-                FileType = ToolBox.ReadString(data, 8, ref pos);
-                if (FileType != "MPLS0100" && FileType != "MPLS0200" && FileType != "MPLS0300")
-                {
-                    throw new Exception(string.Format(
-                        CultureInfo.InvariantCulture,
-                        "Playlist {0} has an unknown file type {1}.",
-                        FileInfo.Name, FileType));
-                }
+            FileType = ToolBox.ReadString(data, 8, ref pos);
+            if (FileType != "MPLS0100" && FileType != "MPLS0200" && FileType != "MPLS0300")
+            {
+                throw new Exception($"Playlist {_fileInfo.Name} has an unknown file type {FileType}.");
```

Comment:
These formats are culture dependent now, I'm not expecting issues, just something worth making a note of